runtime.c

```diff
@@ -247,7 +247,13 @@ portable_option(const char *arg, const char *appimage_path, const char *name)
         char portable_dir[PATH_MAX];
         char fullpath[PATH_MAX];
 
-        readlink(appimage_path, fullpath, sizeof(fullpath));
+        ssize_t length = readlink(appimage_path, fullpath, sizeof(fullpath));
+        if (length < 0) {
+            printf("Error getting realpath for %s\n", appimage_path);
+            exit(EXIT_FAILURE);
+        }
+
+        fullpath[length] = '\0';
```

Comment:
Mind to tell why this appears to be necessary? A char array (remember: arrays are allocated on the stack) should contain `'\0'` elements only.

Reply:
https://linux.die.net/man/2/readlink

`readlink() does not append a null byte to buf. It will truncate the contents (to a length of bufsiz characters), in case the buffer is too small to hold all of the contents.`

Reply:
Interesting. I must've missed that. That means I'll have to adjust code where I use it.